### PR TITLE
added scheduler capable to reconfigure tasks requirements

### PR DESCRIPTION
**Pull request by mtitov**

### CODE REVIEW

File: src/radical/pilot/agent/scheduler/continuous_reconfig.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        super().__init__(cfg, session)
+
+        self._task_reqs = {}  # resource requirements (ranks, cores_per_rank)
+
+    # --------------------------------------------------------------------------
+    #
+    def _set_task_reqs(self):
+
+        reqs_file = ''
```
Comment (mtitov) [inline src/radical/pilot/agent/scheduler/continuous_reconfig.py:30]: provide file name through the pilot environment (set within the PilotDescription) @andre-merzky

Comment (andre-merzky) [reply src/radical/pilot/agent/scheduler/continuous_reconfig.py:30]: ping @andre-merzky

File: src/radical/pilot/agent/scheduler/continuous_reconfig.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            self._task_reqs.update(ru.read_json(reqs_file))
+
+    # --------------------------------------------------------------------------
+    #
+    def work(self, tasks):
+
+        if not self._task_reqs:
+            self._set_task_reqs()
+
+        if self._task_reqs:
```
Comment (andre-merzky) [inline src/radical/pilot/agent/scheduler/continuous_reconfig.py:41]: what happens if that is false - tasks just disappear in that case?

File: src/radical/pilot/agent/scheduler/continuous_reconfig.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+            tasks_reconfigured = []
+            for task in ru.as_list(tasks):
+                for attr in ['ranks', 'cores_per_rank']:
+                    v = int(self._task_reqs.get(attr) or 0)
+                    if v:
```
Comment (andre-merzky) [inline src/radical/pilot/agent/scheduler/continuous_reconfig.py:47]: What happens if this is false - tasks again just disappear?

File: src/radical/pilot/agent/scheduler/continuous_reconfig.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                        task[attr] = v
+                        tasks_reconfigured.append(task)
+
+                if task['cores_per_rank'] > 1:
```
Comment (andre-merzky) [inline src/radical/pilot/agent/scheduler/continuous_reconfig.py:51]: This is quite an assumption - I think we should at least leave existing settings unchanged.

**Comment by andre-merzky** [comment]
@mtitov: please have a look at the current state - does that seem workable for you?

File: src/radical/pilot/agent/scheduler/continuous_reconfig.py (hunk 2 above)
Comment (mtitov) [reply src/radical/pilot/agent/scheduler/continuous_reconfig.py]: reworked

File: src/radical/pilot/agent/scheduler/continuous_reconfig.py (hunk 3 above)
Comment (mtitov) [reply src/radical/pilot/agent/scheduler/continuous_reconfig.py]: reworked

File: src/radical/pilot/agent/scheduler/continuous_reconfig.py (hunk 4 above)
Comment (mtitov) [reply src/radical/pilot/agent/scheduler/continuous_reconfig.py]: agree, keeping it unchanged

**Review by andre-merzky — APPROVED** [review]
LGTM - thanks!

**Pull request merged**